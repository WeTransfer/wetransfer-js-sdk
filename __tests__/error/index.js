const WTError = require('../../src/error');

describe('Custom exception', () => {
  function throwError(message, raw) {
    throw new WTError(message, raw);
  }

  describe('when error object is empty', () => {
    it('should throw an undefined error', () => {
      expect(throwError).toThrowErrorMatchingSnapshot();
    });
  });

  describe('when error message is provided', () => {
    it('should throw the provided message', () => {
      expect(() => throwError('Random error.')).toThrowErrorMatchingSnapshot();
    });
  });

  describe('when error message and response object are provided', () => {
    it('should throw the provided and response messages', () => {
      const error = {
        response: {
          status: 403,
          data: {
            message: 'Nope'
          }
        }
      };
      expect(() =>
        throwError('Random error.', error)
      ).toThrowErrorMatchingSnapshot();
    });
  });

  describe('when response object is provided', () => {
    let error;
    beforeEach(() => {
      error = {
        response: {
          status: 201,
          data: {
            message: 'Nope'
          }
        }
      };
    });

    it('should not append a message if status is 200..299', () => {
      expect(() => throwError('', error)).toThrowErrorMatchingSnapshot();
    });

    it('should append a no retry message if status is 400..499', () => {
      error.response.status = 400;
      expect(() => throwError('', error)).toThrowErrorMatchingSnapshot();
    });

    it('should append a retry message if status is 500..504', () => {
      error.response.status = 500;
      expect(() => throwError('', error)).toThrowErrorMatchingSnapshot();
    });

    it('should append a no idea message if status is not expected', () => {
      error.response.status = 999;
      expect(() => throwError('', error)).toThrowErrorMatchingSnapshot();
    });
  });
});
