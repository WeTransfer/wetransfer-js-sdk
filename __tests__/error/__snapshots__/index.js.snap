// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom exception when error message and response object are provided should throw the provided and response messages 1`] = `
"Random error.
Nope"
`;

exports[`Custom exception when error message is provided should throw the provided message 1`] = `"Random error."`;

exports[`Custom exception when error object is empty should throw an undefined error 1`] = `"Response had a undefined code, no idea what to do with that."`;

exports[`Custom exception when response object is provided should append a no idea message if status is not expected 1`] = `
"Response had a 999 code, no idea what to do with that.
Nope"
`;

exports[`Custom exception when response object is provided should append a no retry message if status is 400..499 1`] = `
"Response had a 400 code, the server will not accept this request even if retried.
Nope"
`;

exports[`Custom exception when response object is provided should append a retry message if status is 500..504 1`] = `
"Response had a 500 code, we could retry.
Nope"
`;

exports[`Custom exception when response object is provided should not append a message if status is 200..299 1`] = `"Nope"`;
